src/pages/Explorer/Detail/BlockInfo.tsx

```diff
@@ -15,51 +28,144 @@ import { BlockStatusIcon, statusText } from "./BlockState"
+          <CopyText className="text-foreground/70" text={block.hash} binary />
+        </div>
       </div>
-      <div className="flex flex-col items-end">
-        <div>Children</div>
-        <BlockChildren hash={block.hash} />
+      <div className="flex flex-wrap items-center gap-3 md:justify-end">
+        <StatusChip state={block.status} />
       </div>
+    </header>
+
+    <div className="grid gap-4 md:grid-cols-2">
+      <DetailTile label="Parent block">
+        <BlockLink hash={block.parent} />
+      </DetailTile>
+      <DetailTile label="Children">
+        <BlockChildren hash={block.hash} />
+      </DetailTile>
     </div>
+
     {block.header && (
-      <div className="text-foreground/80 py-2">
-        <p>
-          State root: {block.header.stateRoot.slice(0, 18)}{" "}
-          <CopyText
-            className="align-middle"
-            text={block.header.stateRoot}
-            binary
-          />
-        </p>
-        <p>
-          Extrinsic root: {block.header.extrinsicRoot.slice(0, 18)}{" "}
-          <CopyText
-            className="align-middle"
-            text={block.header.extrinsicRoot}
-            binary
-          />
-        </p>
+      <div className="grid gap-4 grid-cols-1 md:grid-cols-2">
+        <DetailTile label="Roots">
+          <HashPreview title="state" value={block.header.stateRoot} />
+          <HashPreview title="extrinsic" value={block.header.extrinsicRoot} />
+        </DetailTile>
+        <DetailTile label="Block author">
+          <Author hash={block.hash} header={block.header} />
+        </DetailTile>
       </div>
     )}
+  </section>
+)
+
+const StatusChip: FC<{ state: BlockState }> = ({ state }) => (
+  <span className="inline-flex items-center gap-2 rounded-full border border-foreground/20 bg-foreground/2 px-4 py-2 text-sm font-medium">
+    <BlockStatusIcon size={20} state={state} />
+    {statusText[state]}
+  </span>
+)
+
+const HashPreview: FC<{ title: string; value: string }> = ({
+  title,
+  value,
+}) => (
+  <div className="font-mono text-sm">
+    {title}: {value.slice(0, 18)}…
+    <CopyText className="align-middle" text={value} binary />
   </div>
 )
 
+const DetailTile: FC<{ label: string; children: ReactNode }> = ({
+  label,
+  children,
+}) => (
+  <div className="rounded-xl border border-foreground/10 bg-foreground/2 p-4">
+    <p className="text-xs uppercase tracking-widest text-foreground/60">
+      {label}
+    </p>
+    <div className="mt-2 text-base text-muted-foreground">{children}</div>
+  </div>
+)
+
+const validators$ = state(
+  (block: HexString) =>
+    client$.pipe(
+      switchMap(async (client) => {
+        try {
+          return await client
+            .getTypedApi(polkadot_people)
+            .query.Session.Validators.getValue({
+              at: block,
+            })
+        } catch (ex) {
+          console.error(ex)
+          return null
+        }
+      }),
+    ),
+  null,
+)
```

Comment:
I'd much rather discriminate by `Session.CurrentIndex`.

Session.Validators is around 20kB (600 * 32, in polkadot). I'd say that for every block is a bit too much.

That would be:
- Check Session.CurrentIndex
- If I already have the validators for that one, reuse
- If not, ask for Session.Validators and store cache